### PR TITLE
Delete Tag: Implementing Delete Tag

### PR DESCRIPTION
**Pull request by nbriannl**
**Ongoing PR.**

Modifying delete to work on tags as well.
Eg:
`delete t/word1 t/word2 t/word3`

### CODE REVIEW

**Comment by nbriannl** [comment]
Functionality complete. 
Left some more tests in DeleteCommandParser and DeleteCommand.
Contemplating to refactor deleteTag method in Model to `deleteTag(Set<Tag> setOfTags)` from `deleteTag(Tag tag)`.

File: src/main/java/seedu/address/model/ModelManager.java [hunk 1]
```diff
@@ -58,7 +59,7 @@ public void deleteTag(Tag tag) throws DuplicatePersonException, PersonNotFoundEx
             ReadOnlyPerson oldPerson = addressBook.getPersonList().get(i);
 
             Person newPerson = new Person(oldPerson);
-            Set<Tag> newTags = newPerson.getTags();
+            Set<Tag> newTags = new HashSet<Tag>(newPerson.getTags());
```
Comment (nbriannl) [inline src/main/java/seedu/address/model/ModelManager.java:62]: I fixed a bug here. newPerson.getTags() returns an immutable set.

File: docs/UserGuide.adoc [hunk 2]
```diff
@@ -138,6 +139,25 @@ Deletes the 2nd person in the address book.
...
+
+****
+* You can delete multiple tags with one delete command.
+* The tag (or all the tags) specified must already exist in the address book.
+****
+
+Examples:
+
+* `list` +
+`delete t/friends` +
```
Comment (zacharytang) [inline docs/UserGuide.adoc:156]: Good to specify if the command is case-sensitive regarding the tags?

**Review by zacharytang — CHANGES_REQUESTED** [review]

File: docs/UserGuide.adoc (hunk 2 above)
Comment (nbriannl) [reply docs/UserGuide.adoc:156]: Good point. Made a commit. Please review

**Review by zacharytang — APPROVED** [review]

**Pull request merged**